odl/operator/default_ops.py

```diff
@@ -340,11 +360,29 @@ def adjoint(self):
         >>> op2 = MultiplyOperator(3.0, domain=r3, range=r3)
         >>> op2.adjoint(x)
         rn(3).element([ 3.,  6.,  9.])
+
+        Multipy operator with complex space:
+
+        >>> c3 = odl.cn(3)
+        >>> x_complex = c3.element([1, 1j, 1-1j])
+        >>> op3 = MultiplyOperator(x_complex)
+        >>> op3.adjoint.multiplicand
+        cn(3).element([ 1.-0.j,  0.-1.j,  1.+1.j])
         """
         if self.__domain_is_field:
-            return InnerProductOperator(self.multiplicand)
+            if isinstance(self.domain, RealNumbers):
+                return InnerProductOperator(self.multiplicand)
+            elif isinstance(self.domain, ComplexNumbers):
+                return InnerProductOperator(self.multiplicand.conjugate())
+            else:
+                raise TypeError(
+                    'adjoint only implemented for `RealNumbers` and '
+                    '`ComplexNumbers` when `domain` is a `Field`, got {}'
+                    ''.format(self.domain))
```

Comment:
Too complicated. Just write `'adjoint not implemented for `domain` {!r}'.format(self.domain)`. Also it should be `NotImplementedError`, not `TypeError`.